pilot/pkg/proxy/envoy/v2/eventhandler.go

```diff
@@ -14,11 +14,47 @@
 
 package v2
 
+import v3 "istio.io/istio/pilot/pkg/proxy/envoy/v3"
+
+// DistributionType represents the type of object we are tracking. This is distinct from Envoy's TypeUrl
+// as TypeUrl is versioned, whereas DistributionType is not
+type DistributionType string
+
+const (
+	ClusterDistributionType  DistributionType = "Cluster"
+	ListenerDistributionType DistributionType = "Listener"
+	RouteDistributionType    DistributionType = "Route"
+	EndpointDistributionType DistributionType = "Endpoint"
+	UnknownDistributionType  DistributionType = ""
+)
+
+var AllDistributionTypes = []DistributionType{
+	ClusterDistributionType,
+	ListenerDistributionType,
+	RouteDistributionType,
+	EndpointDistributionType,
+}
+
+func TypeURLToDistributionType(typeURL string) DistributionType {
+	switch typeURL {
+	case ClusterType, v3.ClusterType:
+		return ClusterDistributionType
+	case EndpointType, v3.EndpointType:
+		return EndpointDistributionType
+	case RouteType, v3.RouteType:
+		return RouteDistributionType
+	case ListenerType, v3.ListenerType:
+		return ListenerDistributionType
+	default:
+		return UnknownDistributionType
+	}
+}
+
 // EventHandler allows for generic monitoring of xDS ACKS and disconnects, for the purpose of tracking
 // Config distribution through the mesh.
 type DistributionStatusCache interface {
 	// RegisterEvent notifies the implementer of an xDS ACK, and must be non-blocking
-	RegisterEvent(conID string, xdsType string, nonce string)
-	RegisterDisconnect(s string, urls []string)
-	QueryLastNonce(conID string, xdsType string) (noncePrefix string)
+	RegisterEvent(conID string, distributionType DistributionType, nonce string)
```

Comment:
nit: `EventType` seems more meaningful in the context  of method call `RegisterEvent`?

Reply:
updated